src/raftstore/store/worker/apply.rs

```diff
@@ -459,7 +465,7 @@ pub fn notify_stale_req(term: u64, cb: Callback) {
 }
 
 /// Check if a write is needed to be issued before handle the command.
-fn should_write_to_engine(cmd: &RaftCmdRequest, wb_keys: usize) -> bool {
+fn should_write_to_engine(cmd: &RaftCmdRequest, wb_keys: usize, for_merge_source: bool) -> bool {
     if cmd.has_admin_request() {
         match cmd.get_admin_request().get_cmd_type() {
```

Comment:
Should we guard this check with `for_merge_source`?

Reply:
no, when handle committed entries for merge, it is still possible to handle some admin requests(not in log gap, just not applied in time) that should write to engine.